crates/bevy_math/src/rects/rect.rs

```diff
@@ -277,6 +277,27 @@ impl Rect {
         r
     }
 
+    /// Check if this rectangle contains another rectangle.
+    /// Touching edges count as containing. An empty rectangle never contains anything.
```

Comment:
```suggestion
    /// Touching edges count as containing.

Reply:
Maybe this could be implemented as a `ShapeContains<T>` (or something) trait to support more types. Then you could ditch the current `contains` method and do:
```rust
impl ShapeContains<URect> for Rect { .. }

impl ShapeContains<Vec2> for Rect { .. }
```
etc. 

Or alternatively a `BoxBoundedShape` trait:
```rust
trait BoxBoundedShape {
  fn min(&self) -> Vec2;
  fn max(&self) -> Vec2;
}
```
and change `contains_rect` signature to:
```rust
  pub fn contain(&self, shape: impl BoxBoundedShape) -> bool {
    // ...
  }
```

Just `contains_rect` is fine though.

Reply:
Yeah IMO we start with a simple concrete API, then figure out how to abstract it in follow-ups.